xarray/backends/cfgrib_.py

```diff
@@ -0,0 +1,125 @@
+#
+# Copyright 2017-2018 European Centre for Medium-Range Weather Forecasts (ECMWF).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+#   Alessandro Amici - B-Open - https://bopen.eu
+#
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from . import common
+from .. import core
+
+
+class WrapGrib(common.BackendArray):
+    def __init__(self, backend_array):
+        self.backend_array = backend_array
+
+    def __getattr__(self, item):
+        return getattr(self.backend_array, item)
+
+    def __getitem__(self, item):
+        key, np_inds = core.indexing.decompose_indexer(
+            item, self.shape, core.indexing.IndexingSupport.OUTER_1VECTOR)
+
+        array = self.backend_array[key.tuple]
+
+        if len(np_inds.tuple) > 0:
+            array = core.indexing.NumpyIndexingAdapter(array)[np_inds]
+
+        return array
+
+
+FLAVOURS = {
```

Comment:
Is this configuration dict something that could live in cfgrid rather than xarray?

Reply:
Yes, will do.